include/zephyr/drivers/spi.h

```diff
@@ -152,24 +152,49 @@ extern "C" {
 /**
  * @brief SPI Chip Select control structure
  *
- * This can be used to control a CS line via a GPIO line, instead of
- * using the controller inner CS logic.
+ * This describes how the SPI CS line should be controlled, including whether
+ * to use GPIO or native hardware CS, and timing parameters.
  *
  */
 struct spi_cs_control {
 	/**
 	 * GPIO devicetree specification of CS GPIO.
-	 * The device pointer can be set to NULL to fully inhibit CS control if
-	 * necessary. The GPIO flags GPIO_ACTIVE_LOW/GPIO_ACTIVE_HIGH should be
+	 *
+	 * If set to NULL, hardware driver can attempt to natively control CS
+	 * automatically if it has that capability and the correct pinmux has
+	 * been applied. Otherwise, NULL will inhibit CS control.
+	 *
+	 * The GPIO flags GPIO_ACTIVE_LOW/GPIO_ACTIVE_HIGH should be
 	 * equivalent to SPI_CS_ACTIVE_HIGH/SPI_CS_ACTIVE_LOW options in struct
 	 * spi_config.
 	 */
 	struct gpio_dt_spec gpio;
+
 	/**
-	 * Delay in microseconds to wait before starting the
-	 * transmission and before releasing the CS line.
+	 * Timing configuration.
+	 * For GPIO CS, delay is microseconds to wait before starting
```

Comment:
this was not only meant for gpio cs actually. thus  why the structure was called spi_cs_control, and not spi_gpio_cs_control (for instance)

Reply:
yeah, that was convenient that it was already named generically, I noticed